walkthroughs/week-5-tdd/project/src/test/java/com/google/sps/FindMeetingQueryTest.java

```diff
@@ -270,5 +360,49 @@ public void notEnoughRoom() {
 
     Assert.assertEquals(expected, actual);
   }
+
+  @Test
+  public void noneMandatory(){
+    //Have no mandatory attendees, Just 2 optional
+    //
+    // Events  :
+    // Day     : |--------------------|
+
+    Collection<Event> events = Arrays.asList(
+        new Event("Event 1", TimeRange.fromStartEnd(TimeRange.START_OF_DAY, TIME_0830AM, false),
+            Arrays.asList(PERSON_A)),
+        new Event("Event 2", TimeRange.fromStartEnd(TIME_0900AM, TIME_1000AM, false),
+            Arrays.asList(PERSON_B)));
+
+    MeetingRequest request = new MeetingRequest(new ArrayList<>(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
+    request.addOptionalAttendee(PERSON_B);
+
+    Collection<TimeRange> actual = query.query(events, request);
+    Collection<TimeRange> expected =
+        Arrays.asList(TimeRange.fromStartDuration(TIME_0830AM, DURATION_30_MINUTES),
+            TimeRange.fromStartEnd(TIME_1000AM, TimeRange.END_OF_DAY, true));
+
+    Assert.assertEquals(expected, actual);
+  }
+
+  @Test
+  public void noneMandatoryNoGaps(){
+    //Have no mandatory attendees, Just 2 optional
+    //
+    // Events  :
+    // Day     : |--------------------|
+
+    Collection<Event> events = Collections.emptySet();
+
+    MeetingRequest request = new MeetingRequest(new ArrayList<>(), DURATION_30_MINUTES);
+    request.addOptionalAttendee(PERSON_A);
```

Comment:
no gaps, so would the optionals have events for the whole day?

Reply:
added to javadoc for clarification, no gaps refers to no events for attendees

Reply:
I'm not sure if that makes sense then. If there are optional attendees, but the attendees are free all day, wouldn't we get the WHOLE_DAY as free?

Scenario 2: When there is no mandatory attendees and there is optional attendees with gaps
   * return available times around gaps for optional attendees

The gaps for optional attendees will be the whole day

Reply:
i think "gap" is referring to when they are free